internal/core/labels.go

```diff
@@ -17,12 +18,26 @@ const (
 	LabelVersion   = LabelBase + ".version"
 )
 
+// DefaultLabels returns the standard set of labels which
+// includes LabelSessionID if the reaper is enabled.
 func DefaultLabels(sessionID string) map[string]string {
-	return map[string]string{
-		LabelBase:      "true",
-		LabelLang:      "go",
-		LabelSessionID: sessionID,
-		LabelVersion:   internal.Version,
+	labels := map[string]string{
+		LabelBase:    "true",
+		LabelLang:    "go",
+		LabelVersion: internal.Version,
+	}
+
+	if !config.Read().RyukDisabled {
+		labels[LabelSessionID] = sessionID
+	}
+
+	return labels
+}
+
+// AddDefaultLabels adds the default labels for sessionID to target.
+func AddDefaultLabels(sessionID string, target map[string]string) {
```

Comment:
Same here: #2775 added a very similar method to merge dst and src labels

Reply:
They actually do kind of opposite jobs, this adds defaults not custom so we can't reuse unfortunately.